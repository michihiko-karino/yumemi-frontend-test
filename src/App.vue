<script lang="ts">
import { defineComponent, onMounted, ref } from 'vue';
import { getPrefectures, type PrefecturesResponse, getPopulationCompositionPerYear } from './api/ResasApiClient';
import { getTotalPopulation } from './lib/getTotalPopulation';

export default defineComponent({
  setup() {
    const TITLE = 'フロントエンドコーディング試験';
    const prefectures = ref<PrefecturesResponse['result'] | null>(null);

    const onChange = async (prefCode: number) => {
      const response = await getPopulationCompositionPerYear(prefCode);
      const totalPopulationDataSet = getTotalPopulation(response);

      console.log(totalPopulationDataSet);
    };

    onMounted(async () => {
      prefectures.value = await getPrefectures();
    });

    return { TITLE, prefectures, onChange };
  },
});
</script>

<template>
  <div>
    <h1>{{ TITLE }}</h1>
    <ul>
      <li
        v-for="prefecture in prefectures!"
        :key="prefecture.prefCode"
      >
        <input
          :id="`pref-${prefecture.prefCode}`"
          type="checkbox"
          :value="prefecture.prefCode"
          @change="onChange(prefecture.prefCode)"
        ><label :for="`pref-${prefecture.prefCode}`">{{
          prefecture.prefName
        }}</label>
      </li>
    </ul>
  </div>
  >
</template>

<style scoped></style>
