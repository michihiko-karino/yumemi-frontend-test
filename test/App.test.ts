import { mount } from '@vue/test-utils';
import App from '../src/App.vue';

test('mount component', async () => {
  expect(App).toBeTruthy();

  const wrapper = mount(App);

  expect(wrapper.text()).toContain('フロントエンドコーディング試験');
  expect(wrapper.html()).toMatchSnapshot();
});
