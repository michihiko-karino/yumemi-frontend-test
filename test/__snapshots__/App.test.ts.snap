// Vitest Snapshot v1

exports[`mount component 1`] = `"<h1>フロントエンドコーディング試験</h1>"`;
